### PR TITLE
ao_sndio: use sio_flush() to improve controls responsiveness

### PR DESCRIPTION
**Pull request by brad0**
Use sio_flush() instead of sio_stop() to improve controls responsiveness.

### CODE REVIEW

**Comment by Dudemanguy** [comment]
Looks OK and I assume this helps. Is it worth having a fallback for older versions of sndio? 1.9.0 is quite new.

**Comment by brad0** [comment]
I am not sure what the most appropriate way would be to do that with Meson, but here is my rough attempt at doing so.

**Comment by brad0** [comment]
We've had this in the OpenBSD mpv port for the last 15 months.

It was in response to the responsiveness of seeking through a video.

The function was added to the sndio API to help audio backends where appropriate..
https://github.com/openbsd/src/commit/ec8a3410fbf86966d4e84ce53a01458ea2cd5c80

File: meson.build [hunk 1]
```diff
@@ -860,6 +860,11 @@ features += {'sndio': sndio.found()}
 if features['sndio']
     dependencies += sndio
     sources += files('audio/out/ao_sndio.c')
+
+    if sndio.version().version_compare('>=1.9.0')
+        flags += ['-DSNDIO_SIO_FLUSH']
+        add_project_arguments(flags, language: 'c')
+    endif
```
Comment (Dudemanguy) [inline meson.build:867]: Ah, the current convention is to just add it as a feature. For example, see the rubberband handling. So you could do something like `features += {'sndio-1-9': sndio.version().version_compare('>= 1.9.0')}` and that will create a HAVE_SNDIO_1_9 you can use.

Comment (brad0) [reply meson.build]: Ah, thanks. That is much simpler.

**Comment by brad0** [comment]
There was one line of whitespace I should have removed.

**Pull request merged**